src/olympia/files/models.py

```diff
@@ -431,8 +431,15 @@ def webext_permissions_list(self):
             return []
         try:
             # Filter out any errant non-strings included in the manifest JSON.
-            return [p for p in self._webext_permissions.permissions
-                    if isinstance(p, basestring)]
+            # Remove any duplicate permissions.
+            permissions = [p for p in self._webext_permissions.permissions
+                           if isinstance(p, basestring)]
+            filtered_permissions = set()
+            filtered_permissions_add = filtered_permissions.add
+            return [p for p in permissions if
+                    not (p in filtered_permissions or
+                         filtered_permissions_add(p))]
```

Comment:
I've not tried it, but how does it look if you combine the two generators into one?  So we're not looping through twice.